### PR TITLE
Image sharer task 3

### LINKED ISSUE #3: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [x] New images that are added show up on the homepage.
- [x] These images are persisted if the browser is closed or even if the
  server is restarted.
- [x] Images are not displayed wider than 400px.
- [x] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by revatidamle**
Closes #3 

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -23,4 +23,41 @@ def test_create__invalid_image
       assert_select '#image_link', 'ht://cdn.pixabay.com/photo/2013/08/22/19/18/rose-174817__340.jpg'
     end
   end
+
+  def test_index__check_reverse_order
+    Image.destroy_all
```
Comment (joelbandi) [inline test/controllers/images_controller_test.rb]: I believe our tests should automatically wipe test db after every test run

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -23,4 +23,41 @@ def test_create__invalid_image
...
+    Image.create(title: 'test_1',
+                 link: 'http://power.itp.ac.cn/~jmyang/funny/fun2.jpg')
+    Image.create(title: 'test_2',
+                 link: 'http://power.itp.ac.cn/~jmyang/funny/fun3.jpg')
+    @all_images = Image.all.reverse
+    assert @all_images[0][:created_at] > @all_images[1][:created_at]
```
Comment (joelbandi) [inline test/controllers/images_controller_test.rb]: This is just testing active record / rails and not our code. We should probably do the `assert_select` and test the order of images. We can use the `first-child` CSS selector to grab the first image section.

File: app/views/images/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  <tr>
+    <th>Title</th>
+    <th>Image</th>
+    <th></th>
+  </tr>
+  <% @all_images.each do |image|%>
+      <tr id= <%= 'id_' + image[:title] %>>
+        <td><%=image.title %></td>
+        <td><%= image_tag  image.link, :width => 400, :crop => "pad"%>
+        </td>
```
Comment (joelbandi) [inline app/views/images/index.html.erb]: Line 12 should be Line 11 should be one line

File: app/views/images/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<h4> Listing all images saved </h4>
+<table>
+  <tr>
+    <th>Title</th>
+    <th>Image</th>
+    <th></th>
```
Comment (joelbandi) [inline app/views/images/index.html.erb]: Extra <th> here

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -23,4 +23,41 @@ def test_create__invalid_image
...
+    # Save 2 rows in DB
+    Image.create(title: 'test_1',
+                 link: 'http://power.itp.ac.cn/~jmyang/funny/fun2.jpg')
+    Image.create(title: 'test_2',
+                 link: 'http://power.itp.ac.cn/~jmyang/funny/fun3.jpg')
+    # Redirect to index page
+    get root_url
+    assert_response(:success)
+
+    # Assert correct images are displayed against title
```
Comment (joelbandi) [inline test/controllers/images_controller_test.rb]: I dont think we need this test as its not really testing anything significant that we wrote in the controller.

**Comment by joelbandi** [comment]
Oh and also we should rebase this onto master instead of merging master into this branch and we should clean up the commits :)

File: app/views/images/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  </tr>
+  <% @all_images.each do |image|%>
+      <tr id= <%= 'id_' + image[:title] %>>
+        <td><%=image.title %></td>
+        <td><%= image_tag  image.link, :width => 400, :crop => "pad"%>
```
Comment (grahamotte) [inline app/views/images/index.html.erb]: Nitpick comment: The  hash style `width: 400, crop: "pad"` is preferred over the older `:width => 400, :crop => "pad"`.

Comment (revatidamle) [reply app/views/images/index.html.erb]: 👍

**Pull request merged**